site/app/libraries/Core.php

```diff
@@ -107,6 +107,28 @@ public function disableRedirects() {
         $this->redirect = false;
     }
 
+    /**
+     * Load the Submitty software version configuration.
+     * @throws \Exception
+     */
+    public function loadSubmittyVersionConfig() {
+        $version_path = FileUtils::joinPaths('/', 'usr','local','submitty','config','version.json');
+        if (file_exists($version_path)) {
+          $contents = json_decode(file_get_contents($version_path),true);
+          if ($contents === null ||
+              !isset($contents['most_recent_git_tag']) ||
+              !isset($contents['short_installed_commit'])) {
+            throw new Exception("Error parsing version.json file");
+          }
+          $this->latest_tag = $contents['most_recent_git_tag'];
+          $this->latest_commit = $contents['short_installed_commit'];
+        }
+        else{
+          $message = "Unable to access version configuration file " . $version_path;
+          $this->addErrorMessage($message);
+        }
+    }
```

Comment:
Was there a reason to not load this file as part of the rest of the Config model?